src/network/gossip.rs

```diff
@@ -83,6 +83,7 @@ pub enum GossipEvent<Ctx: SnapchainContext> {
     ),
     SyncReply(InboundRequestId, sync::Response<SnapchainValidatorContext>),
     BroadcastDecidedValue(proto::DecidedValue),
+    ReadNodeSynced(),
```

Comment:
This breaks the abstraction. This level shouldn't care about app level logic. This should be an instruction to the gossip layer, like `SubscribeToDecidedValuesTopic`